### DISCUSSION TITLE
Add PRAGMA foreign_keys = OFF/ON;

### DISCUSSION THREAD

**Discussion by RandomFractals** [duckdb/duckdb#4205] [Ideas]
to enable creating tables without enforcing foreign key constraints when creating and initializing new duckdb database instance.

See SQLite documentation for more info:

https://www.sqlite.org/foreignkeys.html

https://www.sqlite.org/pragma.html#pragma_foreign_keys

**Comment by Tishj** [comment]
I'm a little confused after reading this.

To clarify, this would enable creating tables with FK constraints even when the referenced `table.column` does not satisfy the requirements of a FK constraint (table has to exist, column has to exist, column has to be a PRIMARY KEY) ?
Essentially changing the enforcing of these requirements to be lazy?

I could imagine this being useful when having to create a series of tables, knowing how they have to be linked, but not wanting to be forced into a specific creation order.
With the risk of being met with an error when attempting to scan the column with the FK constraint in a query, if those requirements are not met by then.

Are there other cases where this feature is useful?

**Comment by RandomFractals** [comment]
@Tishj yes, this feature request is mainly for creating new duckdb instance from other data sources without being concerned about the table dependencies order when tables are created and data is inserted.

I ran into this while trying to create duckdb instance from sqlite in this Observable notebook:

https://observablehq.com/@randomfractals/sqlite-to-duckdb?collection=@randomfractals/duckdb

**Comment by RandomFractals** [comment]
With `PRAGMA foreign_keys = OFF` command, we could execute SQL scripts to create tables without failure when referenced tables are not defined yet. One example would be duckdb `CREATE TABLE` and `INSERT` data statements along the lines of this SQLite dump tutorial:

https://www.sqlitetutorial.net/sqlite-dump/

`albums` table could then be initialized with this script before `artists` table is created, granted that sql would have to be converted to duckdb SQL dialect:

![Create albums.sql](https://www.sqlitetutorial.net/wp-content/uploads/2016/05/SQLite-dump-example.png
 "FKs OFF")

Or we could just create the tables with table references without the order of dependencies:

![Create Tables](https://www.sqlitetutorial.net/wp-content/uploads/2016/05/SQLite-dump-structure.png
 "Create Tables")

Currently that is not possible in duckdb and table dependency order must be determined to execute multiple create table statements with foreign key references.

**Comment by x1244** [comment]
When I run `PRAGMA foreign_keys=OFF;`，I get an error.
> D PRAGMA foreign_keys=OFF;
> Error: Catalog Error: unrecognized configuration parameter "foreign_keys"
> Did you mean: "force_compression"

**Comment by sebas-rodri** [comment]
Is there any way to temporarly deactivate the FK constraint. I want to consolidate multiple duckdb instances that all have the same tables.

I couldn't find a way. I sthere any workaround?
